src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(health, name, hidden = false) {
+    super(health, name);
+
+    this.hidden = hidden;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herb) {
+    if (herb instanceof Herbivore && herb.hidden === false) {
+      herb.health -= 50;
+    }
+
+    if (herb.health === 0) {
+      Animal.alive.pop();
+    }
+  }
```

Comment:
Dead `herb` will not always be at the **end** of the array. Replace its method `pop` with `splice` 😉

Reply:
i thought there was no point in biting an animal that had already died =))